PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/common/ErrorMessageTooltip.java

```diff
@@ -0,0 +1,34 @@
+package com.microsoft.azure.sparkserverless.common;
+
+import com.intellij.ui.JBColor;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class ErrorMessageTooltip extends JToolTip{
+    @Nullable
+    private Popup popup;
+
+    public ErrorMessageTooltip(@NotNull JComponent component) {
+        super();
+        setComponent(component);
+        setForeground(JBColor.RED);
+    }
+
+    void hideToolTip() {
+        if (popup != null) {
+            popup.hide();
+        }
+    }
+
+    void showErrorToolTip(@NotNull Point location, @Nullable String errorMsg) {
+        try {
+            setTipText(errorMsg);
+            popup = PopupFactory.getSharedInstance().getPopup(getComponent(), this, location.x, location.y - 20);
+            popup.show();
+        } catch (Exception ignored) {
```

Comment:
Why to ignore all exceptions?

Reply:
`Popup getPopup(Component owner, Component contents, int x, int y)` throws `IllegalArgumentException` when `contents` is null. In our situation this exception happens when `ErrorMessageTooltip` is not initialized

---
In reply to: [199704571](https://github.com/Microsoft/azure-tools-for-java/pull/1747#discussion_r199704571) [](ancestors = 199704571)